pkg/openwsn/sock/openwsn_sock_udp.c

```diff
@@ -0,0 +1,548 @@
+    unsigned count = num;
+
+    do {
+        uint16_t port = IANA_DYNAMIC_PORTRANGE_MIN +
+                        (openrandom_get16b() % num);
+        if (!_sock_port_in_use(port)) {
+            return port;
+        }
+        --count;
+    } while (count > 0);
+    return 0;
+}
+
+void sock_udp_init(void)
+{
+    _udp_socket_list = NULL;
+}
+
+int sock_udp_create(sock_udp_t *sock, const sock_udp_ep_t *local,
+                    const sock_udp_ep_t *remote, uint16_t flags)
+{
+    /* sanity checks */
+    assert(sock);
+    assert(remote == NULL || remote->port != 0);
+    /* check that if set netifs match */
+    if ((local != NULL) && (remote != NULL) &&
+        (local->netif != SOCK_ADDR_ANY_NETIF) &&
+        (remote->netif != SOCK_ADDR_ANY_NETIF) &&
+        (local->netif != remote->netif)) {
+        return -EINVAL;
+    }
+    /* if set netifs should match the only one available in OpenWSN*/
+    if ((local != NULL) &&
+        (local->netif != SOCK_ADDR_ANY_NETIF) &&
+        (local->netif != openwsn_get_netif())) {
+        return -EINVAL;
+    }
+    if ((remote != NULL) &&
+        (remote->netif != SOCK_ADDR_ANY_NETIF) &&
+        (remote->netif != openwsn_get_netif())) {
+        return -EINVAL;
+    }
+
+    /* check local */
+    memset(&sock->gen_sock.local, 0, sizeof(sock_udp_ep_t));
+    if (local != NULL) {
+        uint16_t port = local->port;
+        if (!_sock_valid_af(local->family)) {
+            return -EAFNOSUPPORT;
+        }
+        if (port == 0U) {
+            port = _get_dyn_port();
+            if (port == 0U) {
+                return -EADDRINUSE;
+            }
+        }
+        else if (!(flags & SOCK_FLAGS_REUSE_EP)) {
+            /* a single IPV6 address is possible in OpenWSN, so only check
+               if port is in use */
+            if (_sock_port_in_use(port)) {
+                return -EADDRINUSE;
+            }
+        }
+        memcpy(&sock->gen_sock.local, local, sizeof(sock_udp_ep_t));
+        sock->gen_sock.local.port = port;
+    }
+
+    /* check and set remote */
+    memset(&sock->gen_sock.remote, 0, sizeof(sock_udp_ep_t));
+    if (remote != NULL) {
+        if (!_sock_valid_af(remote->family)) {
+            return -EAFNOSUPPORT;
+        }
+
+        if (!_sock_valid_addr((sock_udp_ep_t *)remote)) {
+            return -EINVAL;
+        }
+        memcpy(&sock->gen_sock.remote, remote, sizeof(sock_udp_ep_t));
+    }
+
+    /* if all ok and local != NULL, listen */
+    if (local != NULL) {
+        /* start mailbox */
+        mbox_init(&sock->mbox, sock->mbox_queue, OPENWSN_SOCK_MBOX_SIZE);
+#ifdef SOCK_HAS_ASYNC
+        sock->async_cb = NULL;
+#endif
+        /* update socket list */
+        LL_PREPEND(_udp_socket_list, sock);
+    }
+
+    /* set flags */
+    sock->gen_sock.flags = flags;
+
+    return 0;
+}
+
+ssize_t sock_udp_send(sock_udp_t *sock, const void *data, size_t len,
+                  const sock_udp_ep_t *remote)
```

Comment:
Minor style nit, because I was scrolling through this

```suggestion
ssize_t sock_udp_send(sock_udp_t *sock, const void *data, size_t len,
                      const sock_udp_ep_t *remote)
```

Reply:
I uncrustified the file and it found a couple more nit-picks :)